sdk/core/azure-core-amqp/src/test/java/com/azure/core/amqp/implementation/ReactorSenderTest.java

```diff
@@ -117,6 +124,38 @@ public void testLinkSize() throws IOException {
         verify(sender, times(1)).getRemoteMaxMessageSize();
     }
 
+    /**
+     * Testing that we can send message with transaction.
+     */
+    @Test
+    public void testSendWithTransaction() {
+        // Arrange
+        final String transactionId = "1";
+        Message message = Proton.message();
+        message.setMessageId("id");
+        message.setBody(new AmqpValue("hello"));
+
+        DeliveryState transactionDeliveryState = getAmqpTransactionState(transactionId);
+
+
+        ReactorSender reactorSender = new ReactorSender(entityPath, sender, handler, reactorProvider, tokenManager,
+            messageSerializer, Duration.ofSeconds(1), new ExponentialAmqpRetryPolicy(new AmqpRetryOptions()));
+        ReactorSender spyReactorSender = spy(reactorSender);
+
+        doReturn(Mono.empty()).when(spyReactorSender).send(any(byte[].class), anyInt(), anyInt(), any(DeliveryState.class));
```

Comment:
Should scope when we return a mock whenever possible. `any*()` is very large in scope. (ie. You can scope it to `eq(transactionDeliveryState)` instead of `any(DeliveryState.class)` so we don't have false positives in our tests.)